### PR TITLE
Remove support for IngressRoute and Routes

### PR DESCRIPTION
**Pull request by fibbs**
As decided in #106 I hereby remove this Helm Chart's support for `IngressRoute` and `Route` objects altogether.

We have decided that as these Ingress Controllers (Traefik and Openshift) also support standard `Ingress` objects nowadays, it is no longer necessary to support those CRDs in our Helm Chart.

Additionally to that, we have another, soon to be merged PR #121 which implements the possibility to render arbitrary Manifests together with this Chart.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/zabbix-community/helm-zabbix/blob/master/CONTRIBUTING.md) signed
- [x] Chart Version bumped

### CODE REVIEW

**Comment by fibbs** [comment]
@aeciopires Haven't you added the creation of Helm-Docs to Github Actions automatic CI/CD? I am asking because it's still noted [here](https://github.com/zabbix-community/helm-zabbix/blob/master/CONTRIBUTING.md) as a requirement to create a pull request. If so, could you please change this documentation, accordingly? Thanks

**Comment by aeciopires** [comment]
Hi @fibbs!

Thanks for your contributition.

The publishing of the helm chart is associated with the manually created tag. Using this approach, I can accumulate many PRs and publish all the changes together. That is why I did not add helm docs to the pipeline to avoid conflicts. I had a lot of problems with this last year.

So, during the process of publishing a new release, I evaluate all the changes from the PRs that I will publish, choose the appropriate version number and then publish them in one package...

The explanation for this is that last year I was very busy at work, so several PRs accumulated throughout the year and I published them all in one batch release.

Another reason is that some people who opened the PR were not responding quickly when I had free time (which is normal because everyone's schedule is asynchronous even depending on their interest and free time).

File: charts/zabbix/Chart.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ---
 apiVersion: v2 # Don't change this
 name: zabbix
-version: 6.0.0 # helm chart version
+version: 6.0.1 # helm chart version
```
Comment (aeciopires) [inline charts/zabbix/Chart.yaml:4]: Hi @fibbs!

Can you remove this version? I will choose the correct number during the publish process.

**Review by aeciopires — CHANGES_REQUESTED** [review]

**Comment by aeciopires** [comment]
Hi @fibbs!

What do you think about using this PR to also remove the values ​​and manifests related to Karpenter?

Looking at your PR and contribution of @sinux-l5d in #121, now it is flexible for anyone to add whatever manifest they want to the helm chart according to their needs without us having to worry about approving it for each release. 

I think that the next release of chart is a break change with this values. I think that 7.0.0 version if the new release is more appropriated. Do you agree?

**Comment by fibbs** [comment]
> The publishing of the helm chart is associated with the manually created tag. Using this approach, I can accumulate many PRs and publish all the changes together. That is why I did not add helm docs to the pipeline to avoid conflicts. I had a lot of problems with this last year.
> 
> So, during the process of publishing a new release, I evaluate all the changes from the PRs that I will publish, choose the appropriate version number and then publish them in one package...
> 
> The explanation for this is that last year I was very busy at work, so several PRs accumulated throughout the year and I published them all in one batch release.

I was actually thinking about exactly the same: not raising versions at PRs anymore and to do that "centrally" by one, you or maybe some day me as well :-) So, we should change the documentation of how to contribute anyway and not mention the helm-docs creation nor the version updating, right?

**Comment by aeciopires** [comment]
I agree @fibbs. Can you adjust the documentation in this PR and remove any reference related to Karpenter in values and manifests?

Other point, can you change the READ.me.gotmpl file explaning the break change related to ingressRoute, Route and Karpenter because the new feature is implemented in #121?

File: charts/zabbix/values.yaml [hunk 2]
```diff
@@ -187,8 +187,8 @@ zabbixServer:
     type: ClusterIP
     # -- clusterIP is the IP address of the service and is usually assigned randomly.
     #If an address is specified manually, is in-range (as per system configuration), and is not in use, it will be allocated to the service.
-    clusterIP:
-    # -- externalTrafficPolicy describes how nodes distribute service traffic they receive on one of the Service's "externally-facing" addresses
+    clusterIP: # -- externalTrafficPolicy describes how nodes distribute service traffic they receive on one of the Service's "externally-facing" addresses
+
```
Comment (aeciopires) [inline charts/zabbix/values.yaml:191]: Hi @fibbs!

Can you undo this change?

File: charts/zabbix/values.yaml [hunk 3]
```diff
@@ -221,7 +221,7 @@ zabbixServer:
     # -- Annotations for the zabbix-server service
     annotations: {}
     # metallb.universe.tf/address-pool: production-public-ips
-  # -- Extra environment variables. A list of additional environment variables. List can be extended with other environment variables listed here: https://github.com/zabbix/zabbix-docker/tree/6.0/Dockerfiles/server-pgsql/alpine#environment-variables. See example: https://github.com/zabbix-community/helm-zabbix/blob/master/charts/zabbix/docs/example/kind/values.yaml
+    # -- Extra environment variables. A list of additional environment variables. List can be extended with other environment variables listed here: https://github.com/zabbix/zabbix-docker/tree/6.0/Dockerfiles/server-pgsql/alpine#environment-variables. See example: https://github.com/zabbix-community/helm-zabbix/blob/master/charts/zabbix/docs/example/kind/values.yaml
```
Comment (aeciopires) [inline charts/zabbix/values.yaml:224]: Hi @fibbs!

Can you undo this change?

File: charts/zabbix/values.yaml [hunk 4]
```diff
@@ -287,13 +285,13 @@ postgresql:
     type: ClusterIP
     # -- clusterIP is the IP address of the service and is usually assigned randomly. If an address is specified manually,
     #is in-range (as per system configuration), and is not in use, it will be allocated to the service.
-    clusterIP:
-    # -- Port of service in Kubernetes cluster
+    clusterIP: # -- Port of service in Kubernetes cluster
```
Comment (aeciopires) [inline charts/zabbix/values.yaml:288]: Hi @fibbs!

Can you undo this change?

File: charts/zabbix/values.yaml [hunk 5]
```diff
@@ -366,8 +364,8 @@ zabbixProxy:
     type: ClusterIP
     # -- clusterIP is the IP address of the service and is usually assigned randomly.
     #If an address is specified manually, is in-range (as per system configuration), and is not in use, it will be allocated to the service.
-    clusterIP:
-    # -- externalTrafficPolicy describes how nodes distribute service traffic they receive on one of the Service's "externally-facing" addresses
+    clusterIP: # -- externalTrafficPolicy describes how nodes distribute service traffic they receive on one of the Service's "externally-facing" addresses
```
Comment (aeciopires) [inline charts/zabbix/values.yaml:367]: Hi @fibbs!

Can you undo this change?
Its a problem of identation.

File: charts/zabbix/values.yaml [hunk 6]
```diff
@@ -400,7 +398,7 @@ zabbixProxy:
     # -- Annotations for the zabbix-proxy service
     annotations: {}
     # metallb.universe.tf/address-pool: production-public-ips
-  # -- Extra environment variables. A list of additional environment variables. List can be extended with other environment variables listed here: https://github.com/zabbix/zabbix-docker/tree/6.0/Dockerfiles/proxy-sqlite3/alpine#environment-variables. See example: https://github.com/zabbix-community/helm-zabbix/blob/master/charts/zabbix/docs/example/kind/values.yaml
+    # -- Extra environment variables. A list of additional environment variables. List can be extended with other environment variables listed here: https://github.com/zabbix/zabbix-docker/tree/6.0/Dockerfiles/proxy-sqlite3/alpine#environment-variables. See example: https://github.com/zabbix-community/helm-zabbix/blob/master/charts/zabbix/docs/example/kind/values.yaml
```
Comment (aeciopires) [inline charts/zabbix/values.yaml:401]: Hi @fibbs!

Can you undo this change?
Its a problem of identation.

File: charts/zabbix/values.yaml [hunk 7]
```diff
@@ -474,8 +472,8 @@ zabbixAgent:
     type: ClusterIP
     # -- clusterIP is the IP address of the service and is usually assigned randomly.
     #If an address is specified manually, is in-range (as per system configuration), and is not in use, it will be allocated to the service.
-    clusterIP:
-    # -- externalTrafficPolicy describes how nodes distribute service traffic they receive on one of the Service's "externally-facing" addresses
+    clusterIP: # -- externalTrafficPolicy describes how nodes distribute service traffic they receive on one of the Service's "externally-facing" addresses
```
Comment (aeciopires) [inline charts/zabbix/values.yaml:475]: Hi @fibbs!

Can you undo this change?
Its a problem of identation.

File: charts/zabbix/values.yaml [hunk 8]
```diff
@@ -508,7 +506,7 @@ zabbixAgent:
     # -- Annotations for the zabbix-agent service
     annotations: {}
     # metallb.universe.tf/address-pool: production-public-ips
-  # -- If true, agent pods mounts host / at /host/root
+    # -- If true, agent pods mounts host / at /host/root
```
Comment (aeciopires) [inline charts/zabbix/values.yaml:509]: Hi @fibbs!

Can you undo this change?
Its a problem of identation.

File: charts/zabbix/values.yaml [hunk 9]
```diff
@@ -583,8 +581,8 @@ zabbixWeb:
     type: ClusterIP
     # -- clusterIP is the IP address of the service and is usually assigned randomly.
     #If an address is specified manually, is in-range (as per system configuration), and is not in use, it will be allocated to the service.
-    clusterIP:
-    # -- externalTrafficPolicy describes how nodes distribute service traffic they receive on one of the Service's "externally-facing" addresses
+    clusterIP: # -- externalTrafficPolicy describes how nodes distribute service traffic they receive on one of the Service's "externally-facing" addresses
```
Comment (aeciopires) [inline charts/zabbix/values.yaml:584]: Hi @fibbs!

Can you undo this change?
Its a problem of identation.

File: charts/zabbix/values.yaml [hunk 10]
```diff
@@ -616,7 +614,7 @@ zabbixWeb:
     # -- Annotations for the Zabbix Web
     annotations: {}
     # metallb.universe.tf/address-pool: production-public-ips
-  # -- Extra environment variables. A list of additional environment variables. List can be extended with other environment variables listed here: https://github.com/zabbix/zabbix-docker/tree/6.0/Dockerfiles/web-apache-pgsql/alpine#environment-variables. See example: https://github.com/zabbix-community/helm-zabbix/blob/master/charts/zabbix/docs/example/kind/values.yaml
+    # -- Extra environment variables. A list of additional environment variables. List can be extended with other environment variables listed here: https://github.com/zabbix/zabbix-docker/tree/6.0/Dockerfiles/web-apache-pgsql/alpine#environment-variables. See example: https://github.com/zabbix-community/helm-zabbix/blob/master/charts/zabbix/docs/example/kind/values.yaml
```
Comment (aeciopires) [inline charts/zabbix/values.yaml:617]: Hi @fibbs!

Can you undo this change?
Its a problem of identation.

File: charts/zabbix/values.yaml [hunk 11]
```diff
@@ -694,13 +692,13 @@ zabbixWebService:
     type: ClusterIP
     # -- clusterIP is the IP address of the service and is usually assigned randomly. If an address is specified manually,
     #is in-range (as per system configuration), and is not in use, it will be allocated to the service.
-    clusterIP:
-    # -- Port of service in Kubernetes cluster
+    clusterIP: # -- Port of service in Kubernetes cluster
```
Comment (aeciopires) [inline charts/zabbix/values.yaml:695]: Hi @fibbs!

Can you undo this change?
Its a problem of identation.

File: charts/zabbix/values.yaml [hunk 12]
```diff
@@ -694,13 +692,13 @@ zabbixWebService:
...
+
     port: 10053
     # -- Annotations for the Zabbix Web Service
     annotations: {}
     # metallb.universe.tf/address-pool: production-public-ips
-  # -- Extra environment variables. A list of additional environment variables. List can be extended with other environment variables listed here: https://github.com/zabbix/zabbix-docker/tree/6.0/Dockerfiles/web-service/alpine#environment-variables. See example: https://github.com/zabbix-community/helm-zabbix/blob/master/charts/zabbix/docs/example/kind/values.yaml
+    # -- Extra environment variables. A list of additional environment variables. List can be extended with other environment variables listed here: https://github.com/zabbix/zabbix-docker/tree/6.0/Dockerfiles/web-service/alpine#environment-variables. See example: https://github.com/zabbix-community/helm-zabbix/blob/master/charts/zabbix/docs/example/kind/values.yaml
```
Comment (aeciopires) [inline charts/zabbix/values.yaml:701]: Hi @fibbs!

Can you undo this change?
Its a problem of identation.

File: charts/zabbix/values.yaml [hunk 13]
```diff
@@ -764,8 +762,8 @@ zabbixJavaGateway:
     type: ClusterIP
     # -- clusterIP is the IP address of the service and is usually assigned randomly.
     #If an address is specified manually, is in-range (as per system configuration), and is not in use, it will be allocated to the service.
-    clusterIP:
-    # -- externalTrafficPolicy describes how nodes distribute service traffic they receive on one of the Service's "externally-facing" addresses
+    clusterIP: # -- externalTrafficPolicy describes how nodes distribute service traffic they receive on one of the Service's "externally-facing" addresses
```
Comment (aeciopires) [inline charts/zabbix/values.yaml:765]: Hi @fibbs!

Can you undo this change?
Its a problem of identation.

File: charts/zabbix/values.yaml [hunk 14]
```diff
@@ -798,7 +796,7 @@ zabbixJavaGateway:
     # -- Annotations for the zabbix-java-gateway service
     annotations: {}
     # metallb.universe.tf/address-pool: production-public-ips
-  # -- Extra environment variables. A list of additional environment variables. List can be extended with other environment variables listed here: https://github.com/zabbix/zabbix-docker/tree/6.0/Dockerfiles/agent2/alpine#environment-variables. See example: https://github.com/zabbix-community/helm-zabbix/blob/master/charts/zabbix/docs/example/kind/values.yaml
+    # -- Extra environment variables. A list of additional environment variables. List can be extended with other environment variables listed here: https://github.com/zabbix/zabbix-docker/tree/6.0/Dockerfiles/agent2/alpine#environment-variables. See example: https://github.com/zabbix-community/helm-zabbix/blob/master/charts/zabbix/docs/example/kind/values.yaml
```
Comment (aeciopires) [inline charts/zabbix/values.yaml:799]: Hi @fibbs!

Can you undo this change?
Its a problem of identation.

**Review by aeciopires — CHANGES_REQUESTED** [review]
Great job @fibbs! 

I suggest to create a new section called:

#  Only code mainteners

To generate a new release of the helm chart do this:

* Review and merge the opened PRs;

* Run local tests;

* Create a branch. Example:

```
git checkout -b BRANCH_NAME
```

Make sure you are on the correct branch using the following command. The branch in use contains the '*' before the name.

```
git branch
```

* Make your changes and tests to the new branch.

* Verify the syntax errors using the follow commands:

```
cd charts/zabbix
make lint
```

* Analyse the changes of the PRs merged and set a new release using the follow aproach:

_"major version" should be the "dot-release", when "minor" is the dot-dot release: 6.0.2 -> 6.0.3 (example) would then be a minor upgrade, no changes in any APIs, interfaces etc. Should be necessary, and 6.0.2 -> 6.1.0 (example) should be a major (where you have to check whether to modify your values.yaml)._ 

* Change the ``version`` and ``appVersion`` parameters (helm chart and Zabbix version respectively) in ``charts/zabbix/Chart.yaml`` and ``charts/zabbix/artifacthub-pkg.yml`` files.

* Run the follow commands to update the documentation of the helm chart.

```
cd charts/zabbix
make gen-docs
```

* Commit the changes to the branch.

* Push files to repository remote with command:

```
git push --set-upstream origin BRANCH_NAME
```

* Create Pull Request (PR) to the master branch. See this [tutorial](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request-from-a-fork)

* Update the content with the suggestions of the reviewer (if necessary).

* After your pull request is merged to the master branch, update your local clone:

```
git checkout master
git pull upstream master
```

* Clean up after your pull request is merged with command:

```
git branch -d BRANCH_NAME
```

Then you can update the master branch in your forked repository.

```
git push origin master
```

And push the deletion of the feature branch to your GitHub repository with command:

```
git push --delete origin BRANCH_NAME
```

* Create a new tag using to generate a new release of the helm chart using the follow commands:

```
git tag -a 6.1.0 -m "New release" #example
git push upstream --tags
```

* The before commands will start the pipeline and will generate a new release and tag in standad ``zabbix-6.0.0``.

* To keep your fork in sync with the original repository, use these commands:

```
git pull upstream master
git pull upstream master --tags


git push origin master
git push origin master --tags
```

* After this, edit and adjust the text generated automatically for new release and adjust the release notes follow the examples the other releases published in https://github.com/zabbix-community/helm-zabbix/releases

# Reference:

* https://blog.scottlowe.org/2015/01/27/using-fork-branch-git-workflow/


What do you think @fibbs? Feel free to adjust the content...

File: charts/zabbix/README.md.gotmpl [hunk 15]
```diff
@@ -142,6 +142,11 @@ helm uninstall zabbix -n monitoring
 
 # Breaking changes of this helm chart
 
+## Version 6.1.0
+* Removing support for non-default Kubernetes features and Custom Resource objects: `IngressRoute`, `Route`, more info: #123
+* Removing support for [karpenter](https://karpenter.sh) due to the more generalistic approach: #121
+* Adding support to deploy any arbitrary manifests together with this Helm Chart by embedding them in the `.Values.extraManifests` list (#121)
+
```
Comment (aeciopires) [inline charts/zabbix/README.md.gotmpl:149]: I think that is necessary add new changes about #127 PR

As for all the other changes. It's excellent. Thanks for the huge contribution.

**Review by aeciopires — CHANGES_REQUESTED** [review]

**Review by aeciopires — APPROVED** [review]
Thanks @fibbs.

Very good

**Pull request merged**